### PR TITLE
Prepare for 0.4

### LINKED ISSUE #53: Replace unwrap in examples with `?`

### PR DESCRIPTION
**Pull request by Nemo157**
Remove non-prepared alphabet support and use new compiler features.

fixes #53

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -43,8 +46,15 @@ impl FromStr for Alphabet {
                 if bytes.len() != 58 {
                     return Err(anyhow!("custom alphabet is not 58 characters long"));
                 }
-                let ptr = bytes.as_ptr() as *const [u8; 58];
-                Alphabet::Custom(unsafe { *ptr })
+                // SAFETY: Length checked just above
+                let bytes = unsafe { &*(bytes.as_ptr().cast::<[u8; 58]>()) };
```
Comment (jyn514) [inline cli/src/main.rs:50]: This is just `transmute`, right? Better to go through transmute so you get the runtime checks that this is valid.

File: src/decode.rs [hunk 2]
```diff
@@ -9,16 +9,16 @@ use crate::Check;
 #[cfg(feature = "check")]
 use crate::CHECKSUM_LEN;
 
-use crate::alphabet::{Alphabet, AlphabetCow};
+use crate::Alphabet;
 
 /// A builder for setting up the alphabet and output of a base58 decode.
 ///
-/// See the documentation for [`bs58::decode`](../fn.decode.html) for a more
+/// See the documentation for [`bs58::decode`](crate::decode()) for a more
```
Comment (jyn514) [inline src/decode.rs:16]: 🎉

File: cli/src/main.rs (hunk 1 above)
Comment (Nemo157) [reply cli/src/main.rs:50]: No, it's going from a `&[u8]` thicc reference to a `&[u8; 58]` thin reference, via getting the `*const u8` pointer to the start of the slice and widening that to a `*const [u8; 58]` pointer to the 58 bytes needed.

Comment (Nemo157) [reply cli/src/main.rs:50]: (Though, if I just wait 2 days I can replace it with the `TryFrom` impl below when 1.47 releases 😁)

**Comment by Nemo157** [comment]
Replaced by #60, #61, #62 and #63

**Pull request closed without merging**